src/Agent.Worker/AgentPluginManager.cs

```diff
@@ -20,96 +18,42 @@ namespace Microsoft.VisualStudio.Services.Agent.Worker
     public interface IAgentPluginManager : IAgentService
     {
         List<string> GetTaskPlugins(Guid taskId);
-        AgentCommandPluginInfo GetPluginCommad(string commandArea, string commandEvent);
         Task RunPluginTaskAsync(IExecutionContext context, string plugin, Dictionary<string, string> inputs, Dictionary<string, string> environment, Variables runtimeVariables, EventHandler<ProcessDataReceivedEventArgs> outputHandler);
-        void ProcessCommand(IExecutionContext context, Command command);
     }
 
     public sealed class AgentPluginManager : AgentService, IAgentPluginManager
     {
-        private readonly Dictionary<string, Dictionary<string, AgentCommandPluginInfo>> _supportedLoggingCommands = new Dictionary<string, Dictionary<string, AgentCommandPluginInfo>>(StringComparer.OrdinalIgnoreCase);
         private readonly Dictionary<Guid, List<string>> _supportedTasks = new Dictionary<Guid, List<string>>();
 
-        private readonly HashSet<string> _taskPlugins = new HashSet<string>()
```

Comment:
Are there any vulnerabilities we open ourselves up to by not loading this list explicitly?

Reply:
if someone compiled the Agent.Plugins. assembly and added in a new plugin task, but then they could have just modified this file as well and changed the list.